.storybook/blocks/ComponentDetails.jsx

```diff
@@ -128,6 +166,188 @@ const VersionDetails = ({ tag, data = {}, isDeprecated = false, skipDate = false
+				<polygon fill="#FFFFFF" points="11,1 11,5 13,5 13,2 14,2 14,5 15,5 15,2 16,2 16,5 17,5 17,1 "></polygon>
+			</svg>
+		</ResourceIconWrapper>
+	)
+};
+
+const githubSvg = () => {
+	return (
+		<ResourceIconWrapper backgroundColor="rgba(0, 0, 0, 0.1)">
+			<svg focusable="false" aria-hidden="true" aria-label="GitHub" viewBox="0 0 36 36">
+				<path fill="rgb(0, 0, 0)" d="M17.988 2a16.291 16.291 0 0 0-5.147 31.747c.814.149 1.111-.354 1.111-.786 0-.386-.014-1.411-.022-2.77-4.531.984-5.487-2.184-5.487-2.184a4.32 4.32 0 0 0-1.809-2.383c-1.479-1.01.112-.99.112-.99a3.42 3.42 0 0 1 2.495 1.679 3.468 3.468 0 0 0 4.741 1.353 3.482 3.482 0 0 1 1.034-2.177C11.4 25.078 7.6 23.68 7.6 17.438a6.3 6.3 0 0 1 1.677-4.371 5.863 5.863 0 0 1 .16-4.311s1.368-.438 4.479 1.67a15.451 15.451 0 0 1 8.157 0c3.109-2.108 4.475-1.67 4.475-1.67a5.857 5.857 0 0 1 .162 4.311 6.286 6.286 0 0 1 1.674 4.371c0 6.258-3.808 7.635-7.437 8.038a3.888 3.888 0 0 1 1.106 3.017c0 2.177-.02 3.934-.02 4.468 0 .436.293.943 1.12.784A16.292 16.292 0 0 0 17.988 2z"></path>
+			</svg>
+		</ResourceIconWrapper>
+	)
+};
+
+const adobeSvg = () => {
+	return (
+		<ResourceIconWrapper>
+			<svg viewBox="0 0 36 36" focusable="false" aria-hidden="true" aria-label="AdobeLogo">
+				<path fill="rgb(250, 15, 0)" d="M22.175 4H34v28L22.175 4zm-8.336 0H2v28L13.839 4zm4.165 10.317l7.538 17.682h-4.939l-2.258-5.632h-5.517l5.176-12.05z"></path>
+			</svg>
+		</ResourceIconWrapper>
+	)
+};
+
+/**
+ * Determines which SVG to use based on the resource link
+ */
+const iconSvgs = (linkType) => {
+	let svgAsset;
+
+	switch(linkType) {
+		case "package": 
+			svgAsset = npmSvg()
+			break;
+		case "repository": 
+			svgAsset = githubSvg()
+			break;
+		case "guidelines": 
+			svgAsset = adobeSvg()
+			break;
+		default: 
+			break;
+	}
+
+	return svgAsset;
+};
+
+/**
+ * Converts the linkType to the subtitle text
+ */
+const convertLinkTypeText = (linkType) => {
+	switch(linkType) {
+		case "package": 
+			linkType = "npm"
+			break;
+		case "repository": 
+			linkType = "GitHub"
+			break;
+		case "guidelines": 
+			linkType = "Spectrum website"
+			break;
+		default: 
+			break;
+	}
+	return linkType;
+};
+
+export const ResourceLinkContent = ({data, linkType=["package", "repository", "guidelines"]}) => {
+	const packageJson = data;
+
+	// componentGuidelinesName
+	let packageName = "";
+	let packageLink = "";
+
+	// componentBetaName: for components with guidelines that are still on spectrum-contributions/beta site
+	let packageAltName = "";
+	let packageAltLink = "";
+
+	// For form and meter, both are nested within other component package.json files
+  let nestedComponent = packageJson?.nestedComponentName ?? undefined;
+	
+	if(linkType === "package") {
+		// NPM package name and link 
+		packageName = packageJson?.name ?? undefined;
+		packageLink = (packageName && typeof packageName !== "undefined") ? `https://npmjs.com/${packageName}` : false;
```

Comment:
The NPM and GitHub links make sense to hardcode but I think we should abstract the guidelines and contributions to just a "designLink" and storing them in the package.json. I think there's a lot of great info we could make available from there.  Let's store it in a `spectrum` section maybe like this:

```
"spectrum": {
	"rootClass": "spectrum-ActionButton",
	"designLink": "https://spectrum.adobe.com/page/action-button",
},
```

Maybe for packages with more than 1 component we could use an array? (@pfulton what do you think?):

```
"spectrum": [{
	"rootClass": "spectrum-Form",
	"designLink": "https://spectrum.adobe.com/page/form",
}, {
	"rootClass": "spectrum-FieldLabel",
	"designLink": "https://spectrum.adobe.com/page/field-label",
}],
```

Reply:
Let me know if this is sort of what you're thinking. I [refactored what I had in all of the `package.json`s](https://github.com/adobe/spectrum-css/pull/3129/commits/c8ce402d3c16937638072a625609346bea714696) to include a `spectrum` array like you suggested. I ended up going with a "guidelinesLink" key within each object, so that when we end up adding Figma links, we can use the "designLink" language for that. If that's overkill, tell me to change it! 😆 I figured Figma and "design" just went better together, rather than our guidelines pages and "design."

I also re-tried getting the nested component stuff to work- maybe it's not a very elegant solution, but I do think it works. It was a little hard to test very thoroughly just because of that weird "undefined.js" thing showing up for progress bar, so I only had the field label/form nesting pair to work against.

https://github.com/adobe/spectrum-css/pull/3129/commits/d6de19b34b738452e5405d591482f221911d8025